openwisp_firmware_upgrader/tasks.py

```diff
@@ -15,6 +16,60 @@
 logger = logging.getLogger(__name__)
 
 
+@shared_task(base=OpenwispCeleryTask, bind=True)
+def delete_firmware_files(self, file_paths):
+    """
+    Deletes firmware files from storage in the background
+    """
+    logger.info(f"Starting delete_firmware_files task with paths: {file_paths}")
+    storage = app_settings.PRIVATE_STORAGE_INSTANCE
+    if not storage:
+        logger.error("No storage instance configured")
+        return
+    logger.info(f"Using storage backend: {storage.__class__.__name__}")
+    logger.info(f"Storage location: {storage.location}")
+
+    for path in file_paths:
+        logger.info(f"Attempting to delete file: {path}")
+        try:
+            try:
+                exists = storage.exists(path)
+                logger.info(f"File exists: {exists}")
+                if exists:
+                    full_path = storage.path(path)
+                    logger.info(f"Full path: {full_path}")
+                    storage.delete(path)
+                    logger.info(f"Successfully deleted file: {path}")
+                else:
+                    logger.warning(f"File does not exist: {path}")
+
+                # Try to delete parent directory if empty
+                dir_path = '/'.join(path.split('/')[:-1])
+                if dir_path:
+                    logger.info(f"Attempting to delete directory: {dir_path}")
+                    try:
+                        if storage.exists(dir_path):
+                            full_dir_path = storage.path(dir_path)
+                            logger.info(f"Full directory path: {full_dir_path}")
+                            if not os.listdir(
+                                full_dir_path
+                            ):  # Check if directory is empty
```

Comment:
Don't use the `os` module here. Use the method of the [storage class](https://docs.djangoproject.com/en/5.1/ref/files/storage/#the-storage-class). The storage could be filesystem storage or an S3 bucket.